src/api/views/payment/invoice_webhook_view.py

```diff
@@ -0,0 +1,102 @@
+        payload = request.body
+
+        try:
+            event = stripe.Webhook.construct_event(
+                payload, sig_header, getenv('STRIPE_INVOICE_WEBHOOK_KEY')
+            )
+        except ValueError as e:
+            # Invalid payload
+            return BackendResponse([str(e)], status=status.HTTP_400_BAD_REQUEST)
+        except stripe.error.SignatureVerificationError as e:
+            # Invalid signature
+            return BackendResponse([str(e)], status=status.HTTP_400_BAD_REQUEST)
+        except stripe.error.StripeError as e:
+            print(str(e))
+            BackendResponse(['Something went wrong communicating with Stripe.', str(e)],
+                            status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+        print(event['type'])
+
+        if event['type'] == 'invoice.finalization_failed':
+            # Retry?
+            invoice = event['data']['object']
+
+        if event['type'] == 'invoice.finalized':
+            # Stripe is now trying to collect the money so the user can leave the garage
+            invoice = event['data']['object']
+            try:
+                invoice.pay()
+            except stripe.error.CardError:
+                # Error gets handled by a 'invoice.payment_failed' event
+                pass
+            return complete_order(invoice.metadata)
+
+        if event['type'] == 'invoice.payment_failed':
+            # These events correspond to a failed automatic charge, we have to send an email with another payment option
+            # to the user
+            invoice = event['data']['object']
+            send_invoice_mail(False, invoice)
+            return BackendResponse(f"Automatic charge failed, sent email to user", status=status.HTTP_200_OK, )
+        if event['type'] == 'invoice.payment_succeeded':
+            # The payment succeeded and the user can be notified
+            invoice = event['data']['object']
+            send_invoice_mail(True, invoice)
+            return BackendResponse(f"Payment succeeded, sent email to user.", status=status.HTTP_200_OK, )
+
+
+
+        return BackendResponse(f"Processed event {event['type']}", status=status.HTTP_200_OK, )
+
+
+def send_invoice_mail(succeeded: True, invoice: stripe.Invoice):
+    user = User.objects.get(pk=invoice.metadata['user_id'])
+
+    if succeeded:
+        msg_plain = render_to_string("payment_succeeded_template.txt", {})
+        msg_html = render_to_string("payment_succeeded_template.html", {})
+    else:
+
+        msg_plain = render_to_string(
+            "invoice_failed_template.txt", {"checkout_url": invoice.hosted_invoice_url}
+        )
+        msg_html = render_to_string(
+            "invoice_failed_template.html", {"checkout_url": invoice.hosted_invoice_url}
+        )
+    send_mail(
+        f"Parking boys payment {'succeeded' if succeeded else 'failed'}",
+        msg_plain,
+        EMAIL_HOST_USER,
+        [user.email],
+        html_message=msg_html,
+    )
```

Comment:
Small nit: you could merge this function and `send_checkout_mail()` into when by adding one parameter. If you would do that, you can maybe move them to `core.utils`.